### ISSUE TITLE
503 The request could not be satisfied.

### ISSUE THREAD

**Issue by Talhafayyaz11** [serverless-nextjs/serverless-next.js#1317]
### Describe the bug
![image](https://user-images.githubusercontent.com/49181609/123649975-9f2f0780-d843-11eb-919f-df039047e2da.png)
I am using **@sls-next/serverless-component@3.1.0-alpha.0**. 503 error appears on the pages where **getServerSideProps**  and **getStaticProps** is used. 
### Actual behavior
503 request error on cloudFront. where  **getServerSideProps**  and **getStaticProps** is used. 

### Expected behavior
The package should serve all the pages.
### Project structure
Currently, I am using monorepo architecture for my application. The structure of the project is as follows:
- **Root**
  - **Next app**
      - next.config.json
      - node_modules
      - serverless.yml
      - package.json
  - **node_modules**
  - **shared.js**
  - **package.json**
  - **cypress**

### Screenshots/Code/Logs

**Serverless.yml** :
![image](https://user-images.githubusercontent.com/49181609/123651717-25981900-d845-11eb-845c-084b739493eb.png)
**next.config.json**:
![image](https://user-images.githubusercontent.com/49181609/123652613-e0c0b200-d845-11eb-984d-c3643d2c0462.png)

### Versions

- OS/Environment: ubuntu debian
- @sls-next/serverless-component version: 3.1.0-alpha.0
- Next.js version: ^10.2.0

### Checklist

- [x] You have reviewed the README and [FAQs](https://github.com/serverless-nextjs/serverless-next.js#faq), which answers several common questions.
- [x] Please first try using the most recent `latest` or `alpha` `@sls-next/serverless-component` [release version](https://github.com/serverless-nextjs/serverless-next.js/releases), which may have already fixed your issue or implemented the feature you are trying to use. Note that the old `serverless-next.js` component and the `serverless-next.js` plugin are deprecated and no longer maintained.

**Comment by davegariepy** [comment]
hi @Talhafayyaz11, could you please check if there are any logs? The lambda@edge logs get output to S3 in the region that the request was made.

**Comment by tom-pavemint** [comment]
@davegariepy @Talhafayyaz11  Talha and I are on the same project. Thank you for the tip that the logs move regions.
I found this error:
"errorMessage": "Cannot find module 'next/dist/next-server/server/node-polyfill-fetch'\nRequire stack:\n- /var/task/pages/experiment/offline-state-test.js\n- /var/task/index.js\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js",

Which associates with the client result of:

503 ERROR
The request could not be satisfied.
The Lambda function associated with the CloudFront distribution is invalid or doesn't have the required permissions. We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.
Generated by cloudfront (CloudFront)
Request ID: xpniXL5H0B2DzkG1BdbDQkcvMVqBgg7S6A-C-efbZeJ2kvMtE1R-Kg==

**Comment by dphang** [comment]
> @davegariepy @Talhafayyaz11 Talha and I are on the same project. Thank you for the tip that the logs move regions.
> I found this error:
> "errorMessage": "Cannot find module 'next/dist/next-server/server/node-polyfill-fetch'\nRequire stack:\n- /var/task/pages/experiment/offline-state-test.js\n- /var/task/index.js\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js",
> 
> Which associates with the client result of:
> 
> 503 ERROR
> The request could not be satisfied.
> The Lambda function associated with the CloudFront distribution is invalid or doesn't have the required permissions. We can't connect to the server for this app or website at this time. There might be too much traffic or a configuration error. Try again later, or contact the app or website owner.
> If you provide content to customers through CloudFront, you can find steps to troubleshoot and help prevent this error by reviewing the CloudFront documentation.
> Generated by cloudfront (CloudFront)
> Request ID: xpniXL5H0B2DzkG1BdbDQkcvMVqBgg7S6A-C-efbZeJ2kvMtE1R-Kg==

I think I saw another similar issue here: https://github.com/serverless-nextjs/serverless-next.js/issues/440. Do you mind posting some snippets of what that page `experiment/offline-state-test.js` does (or if possible share a minimal repro)? I assume it is trying to use node fetch polyfill, but I thought Next.js should bundle it properly into the page / chunks JS files. At least when I tried to use fetch in getServerSideProps in the example e2e test app, it bundles it properly.

A hack/workaround would be to use `postBuildCommands` to copy the missing modules from `next/dist/next-server` into the Lambda, but obviously we'd want to fix it if it's a problem with the component. Also, are these modules present in `.next`?

**Comment by Talhafayyaz11** [comment]
![image](https://user-images.githubusercontent.com/49181609/123698817-f9e25680-d877-11eb-8361-b01a9f957449.png)
Here is the code in experiment/offline-state-test

**Comment by Talhafayyaz11** [comment]
structure of .next 
![Uploading image.png…]()

**Comment by Talhafayyaz11** [comment]
structure of  .serverless_nextjs
![image](https://user-images.githubusercontent.com/49181609/123699150-5e051a80-d878-11eb-86c2-4c236008bc2b.png)

**Comment by dphang** [comment]
@Talhafayyaz11 thanks, I mean the full structure as in what is actually being copied into there. And similar for `.next` as well. It should look like this:

![image](https://user-images.githubusercontent.com/3221524/123699397-18166b00-d814-11eb-9a54-91f0b6b736f8.png)

![image](https://user-images.githubusercontent.com/3221524/123699449-282e4a80-d814-11eb-9dd5-663e9223ebc0.png)

For the test e2e app `next-app` I did try to use `fetch` and the `node-fetch-polyfill` is actually bundled into the chunks in `276.js` so it won't fail:

![image](https://user-images.githubusercontent.com/3221524/123699617-54e26200-d814-11eb-95fc-68b2e40287a5.png)

**Comment by Talhafayyaz11** [comment]
@dbhagen  
![image](https://user-images.githubusercontent.com/49181609/123700084-7164b580-d879-11eb-95f7-d06cd9f5d3a2.png)

**Comment by dphang** [comment]
Great, can you show what's in `.next/serverless/pages` as well? and compare to `.serverless_nextjs/default-lambda`

**Comment by dphang** [comment]
Actually it looks like I was testing the e2e test `next-app` on Next.js 10.2.2, which started adding the chunking and I think serverless trace target was deprecated. In 10.2.0 it is still there.

Still, went back and tested on Next.js 10.2.0, I do see for serverless trace target the `node_modules/next/dist/next-server/server/node-polyfill-fetch` is traced and included correctly in the default lambda, and the page works fine using a fetch call.

**Comment by stephanie-pavemint** [comment]
> Great, I think those are the static (front-end) chunks which look fine to me. Can you show what's in `.next/serverless/pages` as well? and compare to `.serverless_nextjs/default-lambda`. This is what is executed on server side that is causing the 503.

I don't follow what we need to compare or we are missing that folder?  I'm Stephanie and I work with Tom and Talha

**Comment by dphang** [comment]
@stephanie-pavemint ok, I don't think we need that anymore. Basically I wanted to see if `.serverless_nextjs/default-lambda` is including `node_modules` and/or other files correctly. Serverless trace target is supposed to figure out all the dependencies, which is based on your `node_modules` in the repo, but this is not actually generated into `.next` (we have to run it in our build process). Anyway, I think that might be broken, maybe due to the monorepo structure? 

I can try to repro based on this and fix, but an easier way for now might just be to upgrade Next.js to 10.2.3 (or 11). At that version, Next.js got rid of serverless trace in lieu of splitting code into chunks properly in `.next`, so it should be working correctly.

**Comment by stephanie-pavemint** [comment]
> @stephanie-pavemint ok, I don't think we need that anymore. Basically I wanted to see if `.serverless_nextjs/default-lambda` is including `node_modules` and/or other files correctly. Serverless trace target is supposed to figure out all the dependencies, which is based on your `node_modules` in the repo, but this is not actually generated into `.next` (we have to run it in our build process). Anyway, I think that trace might be broken and suspect `node_modules` is not generated, maybe due to the monorepo structure? 
> 
> 
> 
> I can try to repro based on this and fix, but an easier way for now might just be to upgrade Next.js to 10.2.3 (or 11). At that version, Next.js effectively got rid of serverless trace and instead are splitting shared code into chunks in `.next`, so it should be working correctly.
> 
> 
> 
> 


Tom upgraded us to 10.2.3 and same error.

**Comment by dphang** [comment]
I see, though I thought the error might be different though, as the posted one was trying to grab the dependency from `node_modules` (because of serverless trace) and failing:

```
"errorMessage": "Cannot find module 'next/dist/next-server/server/node-polyfill-fetch'\nRequire stack:\n- /var/task/pages/experiment/offline-state-test.js\n- /var/task/index.js\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js",
```

If using the new Next.js versions, I believe all the dependency code (e.g the `node-polyfill-fetch`) is now only bundled into `.serverless_nextjs/default/chunks` instead via webpack and so it should be failing at one of these chunks (e.g when trying to require `../chunks/123.js`, for example).

Unfortunately I'm not sure how to help further, so I think you'd still need to try to follow the error stacktrace and see why it's failing (e.g if still failing to require a module, maybe some file is missing?), and try to compare `.next` and `.serverless_nextjs` to see if there are files missing from `.next` that should be in `.serverless_nextjs` (which is what's deployed). Otherwise, please try to provide a minimal repro and I can debug it.

**Comment by Talhafayyaz11** [comment]
@dphang  so i compared **.next/serverless/chunks** with **.serverless_nextjs/default-lambda/chunks**. These chunks are properly copied with no exception. But I noticed one thing that we have **.next/serverless/pages-manifest.json** but not in  **.serverless_nextjs/default-lambda** can that be an issue. Because page-manifest contains a mapping of the route with files.

**Comment by dphang** [comment]
@Talhafayyaz11 great, thanks for confirming. Yes that is expected that it is not there, but it is used. `pages-manifest` is actually read by this component and we convert it a new manifest based on this and other config. You can see the code here: https://github.com/serverless-nextjs/serverless-next.js/blob/aec0f05f5b9f274d9a848b76d4964015ebf4c2f6/packages/libs/lambda-at-edge/src/build.ts#L722-L730.

**Comment by Talhafayyaz11** [comment]
@dphang  @davegariepy  thanks for your quick response. I resolved this issue by upgrading nextjs to 10.2.3 and then adding useServerlesstrace: false and cleanDotNext: true.

**Comment by Talhafayyaz11** [comment]
But I think the real culprit was useServerlesstrace.
